source/server/connection_handler_impl.cc

```diff
@@ -386,6 +427,30 @@ ConnectionHandlerImpl::ActiveTcpListener::getOrCreateActiveConnections(
   return *connections;
 }
 
+void ConnectionHandlerImpl::ActiveTcpListener::removeFilterChains(
+    const std::list<const Network::FilterChain*>& draining_filter_chains) {
+  // Need to recover the original deleting state
+  bool was_deleting = is_deleting_;
+  is_deleting_ = true;
+  for (const auto* filter_chain : draining_filter_chains) {
+    auto iter = connections_by_context_.find(filter_chain);
+    if (iter == connections_by_context_.end()) {
+      // It is possible when listener is stopping.
+    } else {
+      auto& connections = iter->second->connections_;
+      while (!connections.empty()) {
+        connections.front()->connection_->close(Network::ConnectionCloseType::NoFlush);
+      }
+      // since is_deleting_ is on, we need to manually remove the map value and drive the iterator
```

Comment:
nit: start with capital, end with period, etc. (please check this in future PRs so I don't have to comment on this all the time, thank you!), also merge with comment below in a single comment block.